Assets/EngyneCreations/Trello/Scripts/TrelloAPI.cs

```diff
@@ -0,0 +1,312 @@
+					CatchException($"Unable to populate board from Trello : {request.error}");
+				}
+			}
+		}
+		
+		/// <summary>
+		/// Download all the lists of the selected board and store them.
+		/// </summary>
+		/// <returns>Downloaded list.</returns>
+		private async Task PopulateLists()
+		{
+			_lists = null;
+			
+			if (_currentBoardId == null)
+			{
+				CatchException("Cannot retrieve the lists, there isn't a selected board yet.");
+			}
+
+			using (UnityWebRequest request = UnityWebRequest.Get($"{BOARD_BASE_URL}{_currentBoardId}?key={KEY}&token={TOKEN}&lists=all"))
+			{
+				await request.SendWebRequest();
+
+				if (request.result == UnityWebRequest.Result.Success && request.downloadHandler != null)
+				{
+					var dict = Json.Deserialize(request.downloadHandler.text) as Dictionary<string,object>;
+					_lists = (List<object>)dict["lists"];
+				}
+				else
+				{
+					CatchException($"Unable to retrieve the lists : {request.error}");
+				}
+			}
+		}
+
+		private async Task CreateNewBoard(string boardName)
+		{
+			if (string.IsNullOrEmpty(boardName))
+			{
+				CatchException("Unable to create the new board");
+			}
+			
+			WWWForm post = new WWWForm();
+			post.AddField("name",boardName);
+
+			using (UnityWebRequest request = UnityWebRequest.Post($"{BOARD_BASE_URL}?key={KEY}&token={TOKEN}", post))
+			{
+				await request.SendWebRequest();
+
+				if (request.result != UnityWebRequest.Result.Success)
+				{
+					CatchException($"Unable to create new Trello Board : {request.error}");
+				}
+			}
+		}
+		
+		private async Task CreateNewList(string listName)
+		{
+			if (string.IsNullOrEmpty(listName))
+			{
+				throw new TrelloException("Unable to name the new list.");
+			}
+
+			WWWForm post = new WWWForm();
+			post.AddField("name", listName);
+			post.AddField("idBoard", _currentBoardId);
+
+			using (UnityWebRequest request = UnityWebRequest.Post($"{LIST_BASE_URL}?key={KEY}&token={TOKEN}", post))
+			{
+				await request.SendWebRequest();
+				
+				if (request.result != UnityWebRequest.Result.Success)
+				{
+					CatchException($"Unable to create new Trello List : {request.error}");
+				}
+				
+			}
+		}
+
+		private string GetBoardId(string boardName)
+		{
+			if (string.IsNullOrEmpty(boardName))
+			{
+				return null;
+			}
+			
+			for (int i = 0, count = _boards.Count; i < count; i++)
+			{
+				var board = (Dictionary<string, object>)_boards[i];
+				if ((string)board["name"] == boardName)
+				{
+					return (string)board["id"];
+				}
+			}
+
+			return null;
+		}
+
+		private string GetListId(string listName)
+		{
+			if (string.IsNullOrEmpty(listName))
+			{
+				return null;
+			}
+			
+			for (int i = 0, count = _lists.Count; i < count; i++)
+			{
+				var list = (Dictionary<string, object>)_lists[i];
+				
+				if ((string)list["name"] == listName)
+				{
+					return (string)list["id"];
+				}
+			}
+
+			return null;
+		}
+
+		private void CatchException(string errorMessage)
+		{
+			_onFailure?.Invoke();
```

Comment:
You don't have to handle this now. 

But let's say tomorrow, we want to give the actual error message on the dialogue. How would you set that up? 
Just a thought.

Reply:
We can have onFailure handle a string as a parameter.

The reason I did not do that in the first place is because most of the errorMessages are like not really readable, and if users want to read it, they can read the console